File: test_examples_lib.py
"""Library for testing runtime examples in generated Python stubs."""


def calculate_discount(price, percentage):
    discount = price * (percentage / 100)
    return {
        "original": price,
        "discount": discount,
        "final": price - discount,
    }


def format_address(street, city, zip_code):
    return {
        "line1": street,
        "line2": f"{city}, {zip_code}",
        "full": f"{street}, {city}, {zip_code}",
    }


def sum_array(items):
    return {
        "items": items,
        "total": sum(items),
        "count": len(items),
    }

File: test-examples-lib.js
/**
 * Library for testing runtime examples in generated types.
 * These functions produce predictable output for easy verification.
 */

function calculateDiscount(price, percentage) {
  const discount = price * (percentage / 100);
  return {
    original: price,
    discount: discount,
    final: price - discount,
  };
}

function formatAddress(street, city, zip) {
  return {
    line1: street,
    line2: `${city}, ${zip}`,
    full: `${street}, ${city}, ${zip}`,
  };
}

function sumArray(items) {
  return {
    items: items,
    total: items.reduce((a, b) => a + b, 0),
    count: items.length,
  };
}

module.exports = { calculateDiscount, formatAddress, sumArray };
